File: eBiblioteka/eBiblioteka.Core/Entities/Author.cs

namespace eBiblioteka.Core
{
    public class Author :BaseEntity
    {
        public string FullName { get; set; } = null!;
        public string? Biography { get; set; }
        public int BirthYear { get; set; }
        public int? MortalYear { get; set; }

        public int? PhotoId { get; set; }
        public Photo? Photo { get; set; }

        public int CountryId { get; set; }
        public Country Country { get; set; } = null!;

        public int GenderId { get; set; }
        public Gender Gender { get; set; } = null!;

        public ICollection<Book> Books { get; set; } = null!;
    }
}

File: eBiblioteka/UI/admin_ebiblioteka/lib/models/author.dart
import 'dart:ffi';

import '../models/country.dart';
import '../models/photo.dart';
import 'package:json_annotation/json_annotation.dart';
import '../models/gender.dart';

part 'author.g.dart';

@JsonSerializable()
class Author {
  int? id;
  String? fullName;
  String? biography;
  int? birthYear;
  int? mortalYear;
  int? genderId;
  Gender? gender;
  int? countryId;
  Country? country;
  int? photoId;
  Photo? photo;

  Author(
      this.id,
      this.fullName,
      this.biography,
      this.birthYear,
      this.mortalYear,
      this.genderId,
      this.gender,
      this.country,
      this.countryId,
      this.photoId,
      this.photo
      );

  factory Author.fromJson(Map<String, dynamic> json) => _$AuthorFromJson(json);

  Map<String, dynamic> toJson() => _$AuthorToJson(this);
}
